tests/test_timestamp.py

```diff
@@ -453,41 +453,117 @@ def test_convert_str(self):
             self.assertTrue(isinstance(ts, Timestamp))
             self.assertEqual(ts, t[1])
 
-    def test_convert_tai_sec_nsec(self):
+    def test_convert_sec_nsec(self):
+        """This tests that the conversion to and from TAI second:nanosecond pairs works as expected."""
+
+        tests_ts = [
+            ("0:0", TimeOffset(0, 0), "0:0"),
+            ("0:1", TimeOffset(0, 1), "0:1"),
+            ("-0:1", TimeOffset(0, 1, -1), "-0:1"),
+            ("5", TimeOffset(5, 0), "5:0"),
+            ("5:1", TimeOffset(5, 1), "5:1"),
+            ("-5:1", TimeOffset(5, 1, -1), "-5:1"),
+            ("5:999999999", TimeOffset(5, 999999999), "5:999999999")
+        ]
+
+        for t in tests_ts:
+            ts = TimeOffset.from_sec_nsec(t[0])
+            self.assertEqual(
+                ts,
+                t[1],
+                msg="Called with {} {} {}".format(t[0], t[1], t[2]))
+            ts_str = ts.to_sec_nsec()
+            self.assertEqual(
+                ts_str,
+                t[2],
+                msg="Called with {} {} {}".format(t[0], t[1], t[2]))
+
+    def test_ts_convert_tai_sec_nsec(self):
         """This tests that the conversion to and from TAI second:nanosecond pairs works as expected."""
 
         tests_ts = [
             ("0:0", Timestamp(0, 0), "0:0"),
+            ("0:1", Timestamp(0, 1), "0:1"),
+            ("-0:1", Timestamp(0, 0), "0:0"),
             ("5", Timestamp(5, 0), "5:0"),
             ("5:1", Timestamp(5, 1), "5:1"),
+            ("-5:1", Timestamp(0, 0), "0:0"),
             ("5:999999999", Timestamp(5, 999999999), "5:999999999")
         ]
 
         for t in tests_ts:
-            ts = Timestamp.from_tai_sec_nsec(t[0])
-            self.assertTrue(isinstance(ts, Timestamp))
-            self.assertEqual(ts, t[1])
-            ts_str = ts.to_tai_sec_nsec()
-            self.assertEqual(ts_str, t[2])
+            ts = Timestamp.from_sec_nsec(t[0])
+            self.assertTrue(
```

Comment:
Not worth changing here, but this sort of thing where there are multiple asserts might be a good place to use sub-tests :)